### PR TITLE
Fix/pierreh/update dependencies

### PR DESCRIPTION
**Pull request by phieronymus**
No ticket linked to this issue:

- Update dependencies
- Revert string-query bump (current version is not working on IE11 and edge (even with compilation and polyfills))
- Use another casing library as the current ones are not working on IE11 and edge (even with compilation and polyfills)

### CODE REVIEW

File: test/specs/payload.specs.ts [hunk 1]
```diff
@@ -58,7 +57,7 @@ describe("Payload Helpers", () => {
                 {
                     foo: "bar",
                     foo2: {
-                        bar: new Buffer(["test"]),
+                        bar: Buffer.from("test"),
```
Comment (phieronymus) [inline test/specs/payload.specs.ts:60]: `new Buffer` is deprecated. Not an important change though. It should not impact the tests.

File: test/specs/payload.specs.ts [hunk 2]
```diff
@@ -81,11 +80,12 @@ describe("Payload Helpers", () => {
             },
             foo2: ["bar"],
             foo3: [{ bar: "fizz" }],
-            foo4: new Buffer(["test"]),
+            foo4: Buffer.from("test"),
             foo5: undefined,
             foo6: {
                 bar: undefined,
             },
+            foo7: [{ fooBar: "fizz" }],
```
Comment (phieronymus) [inline test/specs/payload.specs.ts:88]: Added one more test case

File: src/utils/object.ts [hunk 3]
```diff
@@ -5,35 +5,29 @@
...
 
+    const ignoredKeySet = new Set(ignoreKeys);
     return Object.keys(obj || {}).reduce((acc: Record<string, any>, key: string) => {
-        let value = obj[key];
+        const formattedKey = ignoredKeySet.has(key) ? key : transformer(key);
+        const value = isObject(obj[key]) ? transformKeys(obj[key], transformer) : obj[key];
 
-        if (ignoreKeys.includes(key)) {
-            return { ...acc, [key]: value };
-        }
```
Comment (phieronymus) [inline src/utils/object.ts:23]: Ups... Let me update the ignored key set again.

Comment (phieronymus) [reply src/utils/object.ts]: Ok me-proofed-tests added, we should be fine now 🙂

File: package.json [hunk 4]
```diff
@@ -61,29 +61,24 @@
...
-    "uuid": "8.3.1"
+    "typescript": "4.1.3",
+    "uuid": "8.3.2"
   },
   "dependencies": {
-    "camelcase": "^6.2.0",
-    "decamelize": "^4.0.0",
+    "change-case": "^4.1.2",
     "events": "^3.2.0",
-    "flat": "^5.0.2",
```
Comment (phieronymus) [inline package.json:77]: This library is not used.

File: src/api/utils/payload.ts [hunk 5]
```diff
@@ -24,52 +26,35 @@ export function containsBinaryData(data: any): boolean {
...
-                }
+    const appendFormData = (data: unknown, root = "", key = "") => {
+        const formattedKey = keyFormatter(key);
+        const path = !root ? formattedKey : formattedKey ? `${root}.${formattedKey}` : root;
+
+        if (isBlob(data)) {
+            // Blob
+            formData.append(path, data);
+        } else if (Array.isArray(data)) {
+            // Array
```
Comment (laxa88) [inline src/api/utils/payload.ts:42]: unnecessary comments? 🤔

File: src/utils/object.ts [hunk 6]
```diff
@@ -5,35 +5,33 @@
...
 
-        if (isObject(value)) {
-            if (Array.isArray(value)) {
-                value = transformArray(value);
-            } else {
-                value = transformKeys(value, transformer);
-            }
-        }
-
-        acc[transformer(key)] = value;
```
Comment (laxa88) [inline src/utils/object.ts:33]: just wondering, the logic changed here a little, is it an improvement or bugfix? 🤔

**Review by laxa88 — APPROVED** [review]

File: src/api/utils/payload.ts (hunk 5 above)
Comment (phieronymus) [reply src/api/utils/payload.ts:42]: Hum... I jut tried to list the cases as it was not obvious that the last one was for string and buffers. I can remove the comments if you think it is extra though 🙂

File: src/utils/object.ts (hunk 6 above)
Comment (phieronymus) [reply src/utils/object.ts:33]: The function already handles the arrays 🙂 Which is why the logic is still the same. The array handling was repeated.

Also made the code immutable in this function (except for `acc` as per convention)

Comment (phieronymus) [reply src/utils/object.ts:33]: Changed the `obj[key]` back to `value` 🙂 Thanks!

Comment (phieronymus) [reply src/utils/object.ts:33]: The linked test to my first point is the array test:

```
myKey2: [
                {
                    mySubKey21: "Value 4-1",
                    mySubKey22: "Value 4-2",
                    mySubKey23: ["Value 4-3-1", "Value 4-3-2"],
                },
            ],
```

**Review by laxa88 — APPROVED** [review]

File: src/api/utils/payload.ts (hunk 5 above)
Comment (laxa88) [reply src/api/utils/payload.ts:42]: doesn't hurt to keep the comments :meow_party:

**Pull request merged**